Hyperlinks

kahummer opened the pull request:
Should be merged after https://github.com/onaio/gisida-react/pull/218
Fixes: https://github.com/onaio/irfsomalia/issues/889,  https://github.com/onaio/irfsomalia/issues/894, https://github.com/onaio/irfsomalia/issues/897
This pr has changed the structure of our current menu to accommodate the hyperlinks. Some styling tweaks will be desired before we merge.
The configuration had to change to target sub-sector/ sub-category level navigation based on the parent sector as a way of discerning in the event of duplicate `sub-sectors`.
BRCis II is the parent sector.
e.g
```
"hyperLink": {
      "Boundaries": {
        "link": "https://drive.google.com/file/d/1W93Tw3MOhjTEz8h2ycUbY-k6yZXf81Pi/view",
      },
      "BRCiS II Output Indicators": {
        "link": "https://drive.google.com/file/d/1W93Tw3MOhjTEz8h2ycUbY-k6yZXf81Pi/view",
      }
```
More docs on this https://github.com/onaio/gisida-react/blob/hyperlinks/docs/features/menu-legend-links.md
The same specification will be used for the info Icon too. The legendInfo won't require this since this is built at the layer level.
Screenshots of the current implementation
![Screenshot from 2020-08-03 20-09-02](https://user-images.githubusercontent.com/16018123/89208528-39031b80-d5c5-11ea-9784-cfe014552f75.png)


- [x] Have consistent ui with previous implementations

- [x] Legend info and Hyperlink

- [x] Add tests

- [x] Add menu links documentation

- [x] Export unsets inner style

- [x] Handle subsectors inside subsectors

kahummer committed:
Add hyperlink at sub-category level

kahummer committed:
Add hyperlink at unconnected sub-category level

kahummer committed:
Add sector level hyperlink

kahummer commented:
Add links and info description to the legend
The following specification should be used
```
"legendHyperlink": "google.com",
"legendDescription": "Provision of emergency humanitarian assistance to vulnerable populations",
"legendHyperlinkText": "Google doc Report",
```
![Screenshot from 2020-07-30 11-52-57](https://user-images.githubusercontent.com/16018123/88905636-27d1ac00-d25f-11ea-9fb3-417ee3f9f9fd.png)

kahummer committed:
Wip default layer with filters

kahummer committed:
Legend link and info Icon

kahummer committed:
Update info description style

kahummer committed:
Menu style tweaks wip

kahummer committed:
Move legendItems to utils

kahummer committed:
Update legend tests

kahummer committed:
Implement subsector links and info description

kahummer committed:
Legend info icons tweaks

kahummer committed:
Add flags to prevent default events

kahummer committed:
Update LegendUtils test

kahummer committed:
Update menu links

kahummer committed:
Update docs

kahummer committed:
Doc update

KipSigei committed:
Retain map internal css after exporting map

Signed-off-by: Kipchirchir Sigei <arapgodsmack@gmail.com>

kahummer committed:
Close layer on legend

kahummer committed:
Legend stacking bug

kahummer committed:
Refactor parent logic

kahummer committed:
Menu tweaks

kahummer committed:
Update snapshot

kahummer committed:
Update docs

kelvin-muchiri commented:
> The legendInfo won't require this since this is built at the layer level.

Instead of duplicating the description in the hyperlink, why can't we use the same info being used by the legend or vice versa? Or can one enter different descriptions for hyperlink and legend?

kelvin-muchiri commented:
> Add links and info description to the legend
> The following specification should be used
> 
> ```
> "legendHyperlink": "google.com",
> "legendDescription": "Provision of emergency humanitarian assistance to vulnerable populations",
> "legendHyperlinkText": "Google doc Report",
> ```
> 
> ![Screenshot from 2020-07-30 11-52-57](https://user-images.githubusercontent.com/16018123/88905636-27d1ac00-d25f-11ea-9fb3-417ee3f9f9fd.png)

Where is this information added? in the `hyperLink` object or?

kelvin-muchiri commented on docs/features/menu-legend-links.md:
````diff
@@ -0,0 +1,37 @@
+        "description": "desc"
+      }
+```
+#### Legend (layer level) spec
````
Could you please add a layer spec including these new properties so it's a bit clear where to add them in the layer spec?

kelvin-muchiri commented on src/lib/components/Filter/Filter.js:
```diff
@@ -283,15 +283,15 @@ export class Filter extends Component {
-        Actions.filtersUpdated(nextProps.mapId, nextProps.layerObj && nextProps.layerObj.id)
-      );
-    }
+    // if (
```
Can we remove the commented code?

kahummer committed:
Toggle sub-menus externally

kahummer committed:
Render legends based on activeLayerIds

kahummer committed:
Remove legendHypertext & legendHyperlink

kahummer commented:
@kelvin-muchiri Just figured we can use categories for the legend links. In that case, I'll be adding the link description only

kelvin-muchiri committed:
Get menu categories from state

As part of performance enhancement, the following changes were made

1. Menu categories are now read from store instead of being built inside the Menu component
2. We wait for map-1 to load before loading map-2 as this was also contributing to the slow performance

Changes in components Layer, Layers, ConnectedLayers underwent changes because unnecessary properties in the category layer objects were removed. The only properties required for the menu categories are id, label and category

kelvin-muchiri committed:
Solve bug layer not loading

kelvin-muchiri committed:
Rename variable

kelvin-muchiri committed:
Remove implementation to add id for menu groups

The addition of a group id is now done during building of the menu categories in gisida

kelvin-muchiri committed:
Solve bug layer checked by default

kelvin-muchiri committed:
Refactor code

kahummer committed:
Update legend style

kelvin-muchiri committed:
Make use of menu group parent instead of count for ConnectedLayers

The group count was used to make sure group names with the same name under different categories do not open/close if one is opened/closed. The same solution can be achieved by using the newly added parent attribute for each group

kelvin-muchiri committed:
Remove console.log, add build to gitignore

kelvin-muchiri committed:
Refactor Menu and add tests for components/Menu/utils/canAccessLayer

kahummer committed:
Menu connected layers tweaks

kelvin-muchiri committed:
Add tests for Menu uitl methods getAccessibleGroupLayer, getAccessibleCategories

kelvin-muchiri committed:
Update doc strings and add test case for Menu util method getAccessibleCategories

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally

kahummer committed:
Category & menu groups toggle fixes

kelvin-muchiri committed:
Get Menu categories from state

categories received from props required to filtered in order to return only the ones the user has access to. The attempt to improve performance by updating the state in componentDidUpdate worked well, but the if a user navigated away from the Map page and navigated back, the Menu was blank. The workaround was to perform the filtering in the render method. This changes remove the work around and make use of the method getDerivedStateFromProps

kelvin-muchiri committed:
Update failing tests for utils/menuGroupHasVisibleLayers

kelvin-muchiri committed:
Remove debugger

kelvin-muchiri committed:
Do not adjust map height of hasNavBar prop is true in Map

Adjusting the height contributed to the legend sliding down after export and the legend content cut in the exported .png file. These changes make sure the map height is not adjusted, the legend bottom position is adjusted if hasNavBar prop is true.

kahummer committed:
Update state with latest props

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally

kahummer committed:
Legend style tweaks

kahummer committed:
Harmonize Layers and connected layer params

kahummer committed:
Move HyperLinks to a component

kahummer committed:
Legend cleanup

kahummer committed:
Menu refactor and test updates

kahummer committed:
Hyperlink tests and component

kahummer committed:
Summary chart style tweaks

kahummer committed:
Build custom shape legend

kahummer committed:
Move layer type methods to utils

kahummer committed:
Update Layer test

kahummer committed:
Delete empty file

kahummer committed:
Remove unnecessary import

kahummer committed:
Cleanup

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally

kahummer committed:
Merge pull request #220 from onaio/toggle-sectors-externally

Toggle  categories and subcategories externally
